### PR TITLE
Prevent over-matching the sysctl output. Issue 10963

### PR DESCRIPTION
**Pull request by stephenw10**
- [x] Redmine Issue: https://redmine.pfsense.org/issues/10963
- [x] Ready for review

This more accurately matches only the coretemp and acpi sensor data sysctls.

### CODE REVIEW

File: src/usr/local/www/widgets/widgets/thermal_sensors.widget.php [hunk 1]
```diff
@@ -34,7 +34,7 @@
 		$_gb = exec("/sbin/sysctl -q dev.cpu | grep temperature | sort",
 		    $dfout);
 	} else {
-		$_gb = exec("/sbin/sysctl -aq | grep temperature", $dfout);
+		$_gb = exec("/sbin/sysctl -aq | grep temperature:", $dfout);
```
Comment (jim-p) [inline src/usr/local/www/widgets/widgets/thermal_sensors.widget.php:37]: That's still too general for my taste. Try this:

```php
$_gb = exec("/sbin/sysctl -aq | /usr/bin/egrep '(hw.acpi.thermal|dev.cpu).*\.temperature:'", $dfout);
```

Or, perhaps even better:

```php
$_gb = exec("/sbin/sysctl -aq hw.acpi.thermal dev.cpu | /usr/bin/grep 'temperature:'", $dfout);
```

**Review by jim-p — CHANGES_REQUESTED** [review]

File: src/usr/local/www/widgets/widgets/thermal_sensors.widget.php (hunk 1 above)
Comment (stephenw10) [reply src/usr/local/www/widgets/widgets/thermal_sensors.widget.php]: That works for me. The 'a' is unused there though.

File: src/usr/local/www/widgets/widgets/thermal_sensors.widget.php [hunk 2]
```diff
@@ -31,10 +31,10 @@
 
 	$specplatform = system_identify_specific_platform();
 	if ($specplatform['name'] == 'SG-5100') {
-		$_gb = exec("/sbin/sysctl -q dev.cpu | grep temperature | sort",
+		$_gb = exec("/sbin/sysctl -q dev.cpu | /usr/bin/grep temperature | sort",
```
Comment (jim-p) [inline src/usr/local/www/widgets/widgets/thermal_sensors.widget.php:34]: May as well use the same pattern here, too. Plus add the full path to `sort`.

**Review by jim-p — CHANGES_REQUESTED** [review]

File: src/usr/local/www/widgets/widgets/thermal_sensors.widget.php (hunk 2 above)
Comment (stephenw10) [reply src/usr/local/www/widgets/widgets/thermal_sensors.widget.php]: Done.

**Review by jim-p — APPROVED** [review]

**Pull request merged**